src/applications/MachineShopSimulator.java

```diff
@@ -21,171 +20,115 @@
     // methods
     /**
      * move theJob to machine for its next task
-     * 
-     * @return false iff no next task
+     * @return false if no next task
      */
     boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
-        if (theJob.getTaskQ().isEmpty()) {// no next task
+        if (theJob.getTaskQ().isEmpty()) {
             simulationResults.setJobCompletionData(theJob.getId(), timeNow, timeNow - theJob.getLength());
             return false;
-        } else {// theJob has a next task
-                // get machine for next task
+        }
+        else {
             int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
-            // put on machine p's wait queue
             machine[p].getJobQ().put(theJob);
             theJob.setArrivalTime(timeNow);
-            // if p idle, schedule immediately
-            if (eList.nextEventTime(p) == largeTime) {// machine is idle
-                changeState(p);
+            if (eList.nextEventTime(p) == largeTime) {
+                Machine.changeState(p, machine[p], eList, timeNow);

```

Comment:
Would it make more sense to make `changeState` a non-static method called on the desired machine?

```suggestion
                machine[p].changeState(p, eList, timeNow);
```